### PR TITLE
Updated old docs in prep for 0.6.0

### PR DESCRIPTION
**Pull request by kdavis-mozilla**

### CODE REVIEW

File: doc/Geometry.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@ n_steps
 -------
 The network views each speech sample as a sequence of time-slices :math:`x^{(i)}_t` of
 length :math:`T^{(i)}`. As the speech samples vary in length, we know that :math:`T^{(i)}`
-need not equal :math:`T^{(j)}` for :math:`i \ne j`. For each batch, BRNN in TensorFlow needs
+need not equal :math:`T^{(j)}` for :math:`i \ne j`. For each batch, RNN in TensorFlow needs
 to know ``n_steps`` which is the maximum :math:`T^{(i)}` for the batch.
```
Comment (reuben) [inline doc/Geometry.rst]: We should remove this, we use dynamic RNNs in the training graph with no need for a fixed or known `n_steps` dimension.

File: doc/DeepSpeech.rst [hunk 2]
```diff
@@ -14,19 +21,19 @@ Let a single utterance :math:`x` and label :math:`y` be sampled from a training
 Each utterance, :math:`x^{(i)}` is a time-series of length :math:`T^{(i)}`
 where every time-slice is a vector of audio features,
 :math:`x^{(i)}_t` where :math:`t=1,\ldots,T^{(i)}`.
-We use MFCC as our features; so :math:`x^{(i)}_{t,p}` denotes the :math:`p`-th MFCC feature
-in the audio frame at time :math:`t`. The goal of our BRNN is to convert an input
+We use MFCC coefficients as our features; so :math:`x^{(i)}_{t,p}` denotes the :math:`p`-th MFCC feature
```
[the remark this answers is not in the document]
Comment (kdavis-mozilla) [reply doc/DeepSpeech.rst]: I'd seen it used this way, despite the fact that, yes, it doesn't make much sense. But that said I can change it.

File: doc/Geometry.rst (hunk 1 above)
Comment (kdavis-mozilla) [reply doc/Geometry.rst]: OK

File: doc/DeepSpeech.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (reuben) [reply doc/DeepSpeech.rst]: Your call. I'm fine with either option.

**Comment by kdavis-mozilla** [comment]
@reuben I addressed review comments. Could you take another look?

**Comment by kdavis-mozilla** [comment]
OK I'll merge as you've already approved.

**Pull request merged**